lib/ansible/plugins/connection/ssh.py

```diff
@@ -505,7 +505,7 @@ def _run(self, cmd, in_data, sudoable=True):
             # and we've read all available output from it, we're done.
 
             if p.poll() is not None:
-                if not rpipes or not rfd:
+                if not rpipes or timeout == 0:
```

Comment:
I'm not sure we need to remove `not rfd`, also existing when timeout is 0 prevents our 'last ditch' effort to read anything left in the buffer.